src/Crank/CoreWCFPerfClient/Program.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Microsoft.Crank.EventSources;
+using System;
+using System.Diagnostics;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+
+namespace CoreWCFPerf
+{
+    public class Parameters
+    {
+        public const string Binding = "binding";
+        public const string ServiceUrl = "serviceurl";
+        public const string ReportingUrl = "reportingurl";
+        public const string PerfMeasurementDuration = "perfmeasurementduration";
+        public const string Requests = "requests";
+        public const string CloseAndOpen = "closeandopen";
+    }
+
+    public enum TestBinding { BasicHttp, NetTcp }
+
+    class Program
+    {
+        private TestBinding _paramBinding = TestBinding.BasicHttp;
+        private TimeSpan _paramPerfMeasurementDuration = s_defaultPerfMeasurementDuration;
+        private string _paramServiceUrl = "";
+        private int _paramRequests = 0;
+        private bool _paramCloseAndOpen = false;
+
+        private readonly static TimeSpan s_defaultPerfMeasurementDuration = TimeSpan.FromSeconds(10);
+
+        static void Main(string[] args)
+        {
+
+            Program test = new Program();
+
+            if (test.ProcessRunOptions(args))
+            {
+                var startTime = DateTime.Now;
+                int requestTime = 0;
+                int request = 0;
+
+                Binding binding = null;
+                switch (test._paramBinding)
+                {
+                    case TestBinding.BasicHttp:
+                        binding = new BasicHttpBinding();
+                        break;
+                    case TestBinding.NetTcp:
+                        binding = new NetTcpBinding(SecurityMode.None);
+                        break;
+                    default:
+                        break;
+                }
+
+                TimeSpan measurementDurationPerTime = test._paramPerfMeasurementDuration ;
+               
+                BenchmarksEventSource.Log.Metadata("firstrequest", "max", "max", "First Request (ms)", "Time to first request in ms", "n0");
+                BenchmarksEventSource.Log.Metadata("corewcfperf/requests", "max", "sum", "Requests (" + test._paramPerfMeasurementDuration.TotalMilliseconds + " ms)", "Total number of requests", "n0");
+                BenchmarksEventSource.Log.Metadata("corewcfperf/rps/max", "max", "sum", "Requests/sec (max)", "Max requests per second", "n0");
+
+                startTime = DateTime.Now;
+                ChannelFactory<ISayHello> factory = new ChannelFactory<ISayHello>(binding, new EndpointAddress(test._paramServiceUrl));
+                factory.Open();
+                var client = factory.CreateChannel();
+                var stopwatchFirstReq = new Stopwatch();
```

Comment:
Client should be opened before first request, otherwise the connection handshake is included in the timing of the first client call. This can be done with this code:
```c#
((IClientChannel)client).Open();
```